internal/webhooks/cluster.go

```diff
@@ -246,6 +247,25 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 		)
 	}
 
+	// upgrade concurrency should be a numeric value.
+	if concurrency, ok := newCluster.Annotations[clusterv1.ClusterTopologyUpgradeConcurrencyAnnotation]; ok {
+		concurrencyAnnotationField := field.NewPath("annotations").Child(clusterv1.ClusterTopologyUpgradeConcurrencyAnnotation)
```

Comment:
```suggestion
		concurrencyAnnotationField := field.NewPath("metadata", "annotations", clusterv1.ClusterTopologyUpgradeConcurrencyAnnotation)
```
nit: sorry missed this on the first review